src/header/parsing.rs

```diff
@@ -9,7 +9,7 @@ pub fn from_one_raw_str<T: str::FromStr>(raw: &[Vec<u8>]) -> Option<T> {
         return None;
     }
     // we JUST checked that raw.len() == 1, so raw[0] WILL exist.
-    if let Ok(s) = str::from_utf8(&raw[0][..]) {
+    if let Ok(s) = str::from_utf8(& unsafe { raw.get_unchecked(0) }[..]) {
```

Comment:
LLVM almost certainly elides the bounds check here anyway. I'd personally rather have less unsafe here. I doubt this has a noticeable impact on perf, but I'd be thrilled to be proven wrong.